### PR TITLE
feat: add support for Starship

### PR DESCRIPTION
**Pull request by foray1010**

### CODE REVIEW

File: mackup/applications/starship.cfg [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[application]
+name = Starship
+
+[configuration_files]
+.config/starship.toml
```
Comment (lumaxis) [inline mackup/applications/starship.cfg:5]: I believe this should use the [built-in XDG support](https://github.com/lra/mackup/tree/master/doc#add-support-for-an-application-using-the-xdg-directory) instead: 
```suggestion
[xdg_configuration_files]
starship.toml
```

Comment (foray1010) [reply mackup/applications/starship.cfg:5]: seems starship is not using XDG: https://github.com/starship/starship/issues/896

**Comment by foray1010** [comment]
already supported in https://github.com/lra/mackup/blob/master/mackup/applications/starship.cfg, closing

File: mackup/applications/starship.cfg (hunk 1 above)
Comment (lumaxis) [reply mackup/applications/starship.cfg:5]: Oh, indeed. I just assumed it was since the file is in `.config`. Sorry about that!

Comment (lra) [reply mackup/applications/starship.cfg:5]: Yeah it's super confusing.

**Pull request closed without merging**